MetaMask/metamask-extension
# Update version parsing to allow rollback release

## Explanation

When we want to rollback a release on Chrome, sometimes we use the fourth part of the version for the rollback release. This is because the Chrome web stores does not directly allow rolling back, but instead requires us to re-submit the release we want to roll back to with a higher version number.

The manifest version parsing now allows for a fourth version part.

The comments have also been updated to be more descriptive, and to fix a minor inaccuracy.

## Manual testing steps

- Try adding a fourth part to the version number in `package.json` (e.g. "v10.12.2.1").
- Ensure that the extension can connect to dapps, that there are no background console errors, and that the "About" page renders the version properly.
